verilog/tools/kythe/kythe_facts_extractor.h

```diff
@@ -0,0 +1,157 @@
+// Copyright 2017-2020 The Verible Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef VERIBLE_VERILOG_TOOLS_KYTHE_KYTHE_FACTS_EXTRACTOR_H_
+#define VERIBLE_VERILOG_TOOLS_KYTHE_KYTHE_FACTS_EXTRACTOR_H_
+
+#include <string>
+#include <utility>
+
+#include "verilog/tools/kythe/indexing_facts_tree.h"
+#include "verilog/tools/kythe/indexing_facts_tree_extractor.h"
+
+namespace verilog {
+namespace kythe {
+
+namespace {
+// Type that is used to keep track of the path to the root of indexing facts
+// tree.
+using IndexingFactsTreeContext2 = std::vector<const IndexingFactNode*>;
+
+using AutoPop2 =
+    AutoPopBack<IndexingFactsTreeContext2, const IndexingFactNode*>;
+
+}  // namespace
+
+// Node vector name for kythe facts.
+struct VName {
```

Comment:
Should this definition come from a Kythe header?

Reply:
I think yes, It's a kythe specific name as kythe uses VNames for node id.

Reply:
@fangism correct me if I am wrong.